tests/integration/cattletest/core/test_volume.py

```diff
@@ -11,3 +11,73 @@ def test_volume_delete_active(client, context):
 
     volume = client.wait_success(client.delete(volume))
     assert volume.state == 'removed'
+
+
+def test_volume_create_state(client, context):
+    name = random_str()
+    c = client.create_volume(name=name, volumeDriver='convoy')
+    c = client.wait_success(c)
+    assert c.state == 'requested'
+
+    assert c.uri == 'convoy:///%s'%name
+
+    volume = client.wait_success(client.delete(c))
+    assert volume.state == 'removed'
+
+
+def test_create_container_with_volume(client, context):
+    name1 = random_str()
+    v1 = client.create_volume(name=name1, volumeDriver='convoy')
+    v1 = client.wait_success(v1)
+    assert v1.state == 'requested'
+
+    name2 = random_str()
+    v2 = client.create_volume(name=name2, volumeDriver='convoy')
+    v2 = client.wait_success(v2)
+    assert v2.state == 'requested'
+
+    dataVolumeMounts = {
+                           '/var/lib/docker/mntpt1': v1.id,
+                           '/var/lib/docker/mntpt2': v2.id
+                       }
+    dataVolumes = {
+                      '/home': '/home'
+                  }
+    c = client.create_container(imageUuid=context.image_uuid, volumeDriver='convoy',
+                                dataVolumes=dataVolumes,
+                                dataVolumeMounts=dataVolumeMounts)
+    c = client.wait_success(c)
+    assert c.state == 'running'
+
+    dataVol1 = '%s:/var/lib/docker/mntpt1'%name1
+    dataVol2 = '%s:/var/lib/docker/mntpt2'%name2
+
+    dataVol1Found = False
+    dataVol2Found = False
+
+    for dataVol in c.dataVolumes:
+        if dataVol == dataVol1:
+            dataVol1Found = True
+        if dataVol == dataVol2:
+            dataVol2Found = True
+
+    assert dataVol1Found and dataVol2Found
+
+    v1 = client.wait_success(v1)
+    v2 = client.wait_success(v2)
+
+    assert v1.state == 'active'
+    assert v2.state == 'active'
+
+    vol1Found = False
+    vol2Found = False
+
+    for vol in c.volumes():
+        if vol.id == v1.id:
+            vol1Found = True
+        if vol.id == v2.id:
+            vol2Found = True
+
+    assert not vol1Found and not vol2Found
```

Comment:
Need to assert that the volume got allocated to a storagePool.  Look for the volumeStoragePoolMap type using the super_client.